content/blog/intro-to-step-functions/index.md

```diff
@@ -0,0 +1,178 @@
+---
+title: "Intro to AWS Serverless Step Functions"
+date: 2020-02-13
+meta_desc: "An Introduction to building serverless apps with AWS Step Functions."
+meta_image: meta-step-function.png
+authors:
+    - sophia-parafina
+tags:
+    - serverless
+    - "step functions"
+---
+
+AWS Step Functions let you build applications by connecting AWS services. We'll go into the details of creating a lambda function, IAM roles and policies, and creating a workflow. Once we have the example deployed, we'll walk through the process of adding another function and step to the workflow. Included in the walkthrough is a discussion of one of the aspects of the Pulumi programming model. The goal of this article is to provide a foundation for building your application using serverless workflows.
```

Comment:
After reading the post I'm still not sure what a "Step function" is. Is it a separate AWS product offering specifically for making it easier to build/manage these workflows? And we use AWS Lambda as a simple example of a "step"?

It would be nice if this was called out a little more explicitly. Since later in the text we just have "The state machine controls the workflow.", and IMHO we could make it a little clearer by elaborating on what we mean by "workflow" and how later, we'll write code to manage the progress through that workflow using a "state machine".

Reply:
I updated the intro paragraph:

[AWS Step Functions](https://aws.amazon.com/step-functions/) lets you build applications by connecting AWS services. Daisy chaining steps into a workflow simplifies application development by creating a state machine diagram which shows how services are connected to each other in your application. We'll go into the details of creating a lambda function, IAM roles and policies, and creating a workflow. Once we have the example deployed, we'll walk through the process of adding another function and step to the workflow. Included in the walkthrough is a discussion of one of the aspects of the Pulumi programming model. The goal of this article is to provide a foundation for building your application using serverless workflows.

Reply:
At least for me, that's a lot clearer in framing what the post is all about. Thanks.